src/apps/_experiments/weather.cpp

```diff
@@ -0,0 +1,807 @@
+        time_current = localtime(&timestamp);
+        if (time_current->tm_mday != mday_prev){
+            strftime(date_buff, sizeof(date_buff), "%d/%m", localtime(&timestamp));
+            dayFirstUpdt.push_back(i);
+        }
+    }
+}
+
+
+void OswAppWeather::printDate(){
+    // //TODO: multiple days-> scrolling menu  
+    time_t time;
+    char date_buf[6];
+    this->hal->gfx()->fillCircle(4,120,3,rgb888(255,255 , 255));
+    if(this->dayFirstUpdt.size()!=0){
+        if(this->updtSelector < this->dayFirstUpdt[1]){
+            this->hal->gfx()->setFont(nullptr);
+            this->hal->gfx()->setTextCursor(40 , 91);
+            this->hal->gfx()->print("---");
+        }else{
+            this->hal->gfx()->setFont(&DS_DIGI12pt7b); 
+            this->hal->gfx()->setTextCursor(40 , 91);
+            time = this->updtTime - 86400;
+            strftime(date_buf, sizeof(date_buf), "%d/%m", localtime(&time));
+            this->hal->gfx()->print(date_buf);
+        }
+        
+        if(this->updtSelector >= this->dayFirstUpdt[this->dayFirstUpdt.size()-1]){
+            this->hal->gfx()->setFont(nullptr);
+            this->hal->gfx()->setTextCursor(40 , 149);
+            this->hal->gfx()->print("---");
+        }else{
+            this->hal->gfx()->setFont(&DS_DIGI12pt7b); 
+            time = this->updtTime + 86400;
+            this->hal->gfx()->setTextCursor(40 , 149);
+            strftime(date_buf, sizeof(date_buf), "%d/%m", localtime(&time));
+            this->hal->gfx()->print(date_buf);
+        }
+        if(this->dayFirstUpdt.size()!=0){
+            this->hal->gfx()->setFont(&DS_DIGI12pt7b); 
+            this->hal->gfx()->setTextCursor(40 , 120);
+            time = this->updtTime;
+            strftime(date_buf, sizeof(date_buf), "%d/%m", localtime(&time));
+            this->hal->gfx()->print(date_buf);
+        }
+    }else{
+        this->hal->gfx()->setFont(nullptr); 
+        this->hal->gfx()->setTextCursor(40 , 120);
+        this->hal->gfx()->print("no data");
+    }
+    
+    this->hal->gfx()->setFont(&DS_DIGI12pt7b);  
+
+
+    // this->hal->gfx()->setTextCursor(40 , 91);
+    if(this->mainSelector==0){
+        //this->hal->gfx()->drawHLine(10,132,58,rgb888(255,255,255));
+        this->hal->gfx()->drawThickLine(10,132,68,132,2,rgb565(164, 35, 52));
+    }
+}
+
+bool OswAppWeather::loadData(){ 
+    String wstr = this->pref.getString("wtr");
+    if (!wstr.equals("")){
+      Serial.println("size of wstr: ");
+      Serial.println(wstr.length());
+      Serial.println("....");
+      //TODO: test decoded data
+      if( (wstr.length() % 8) != 0 ){this->dataLoaded = false; return false;}
+      if( wstr.length()<16){this->dataLoaded = false; return false;}
+      WeatherDecoder decoder(wstr.c_str());
+      this->initTimestamp = decoder.getTime();
+      tm* tmx;
+      this->getDayList();
+      tmx = localtime(&this->initTimestamp);
+      Serial.printf("day :%d\n",tmx->tm_mday);
```

Comment:
You mean `Serial.println()`?

Reply:
Maybe
```
Serial.println("day : " + tmx->tm_mday);
```

Reply:
I think that i should remove some serial prints and leave only the important ones ( errors )